Content/ua_20200713_Feats-2020.js

```diff
@@ -8,4 +8,306 @@ SourceList["UA:F20"] = {
 	group : "Unearthed Arcana",
 	url : "https://media.wizards.com/2020/dnd/downloads/UA2020_Feats.pdf",
 	date : "2020/07/13"
+};
+
+FeatsList["magic initiate [artificer]"] = {
+	name : "Magic Initiate [Artificer]",
+	source : [["UA:F20", 1]],
+	description : "I learn two cantrips and one 1st-level spell of my choice from the artificer's spell list. I can cast the 1st-level spell at its lowest level once per long rest without using a spell slot. Int is my spellcasting ability. I gain proficiency in one artisan's tool that I can use as focus.",
+	spellcastingBonus : [{
+		name : "Artificer cantrip",
+		spellcastingAbility : 4,
+		'class' : 'artificer',
+		level : [0, 0],
+		atwill : true,
+		times : 2
+	}, {
+		name : "Artificer 1st-level spell",
+		'class' : 'artificer',
+		level : [1, 1],
+		oncelr : true
+    }],
+    toolProfs : [["Artisan's tools", 1]]
+};
+FeatsList["chef"] = {
+	name : "Chef",
+	source : [["UA:F20", 1]],
+    description : "During a short rest, I can make food for 4 + my proficiency bonus creatures; if they eat it and spend 1+ Hit Die they regain 1d8 HP. During a long rest I can make snacks equal to my proficiency bonus that give my proficiency bonus temp HP [+1 Constitution or Wisdom]",
+    toolProfs : ["Chef's utensils"],
+    improvements : "Chef (feat): +1 Constitution or Wisdom;"
+};
+FeatsList["crusher"] = {
+	name : "Crusher",
+	source : [["UA:F20", 1]],
+    description : "Once per turn when I deal bludgeoning damage to a target, I can move it 5 feet if it is less than 1 size larger than me. If I deal bludgeoning damage on a critical hit to a target attack rolls against it are with advantage until end of my next turn. [+1 Strength or Dexterity]",
+    improvements : "Crusher (feat): +1 Strength or Dexterity;"
+};
+// NOT DONE //
+FeatsList["eldritch adept"] = {
+	name : "Eldritch Adept",
+	source : [["UA:F20", 1]],
+	description : "I learn one Eldritch Invocation option of my choice from the Warlock class. If it has a prerequisite, I can only take it if I am a warlock and meet the prerequisite. I can replace the invocation whenever I gain a level.",
+    prerequisite : "Spellcasting or Pact Magic feature",
+	prereqeval : "What('Class Features')).toLowerCase().indexOf('spellcasting') === -1) || What('Class Features')).toLowerCase().indexOf('pact magic') === -1)"
```

Comment:
Please use a function, rather than a string that is `eval`ed.

Reply:
This was mostly a guess as to the correct way to identify whether someone has the Spellcasting or Pact Magic features. Could you provide guidance here?

Reply:
Hmm, will consider. I think this is probably fine, just as a function rather than a string. Will investigate.